channeldb/announcement_nonces.go

```diff
@@ -0,0 +1,148 @@
+package channeldb
+
+import (
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec/v2/schnorr/musig2"
+	"github.com/lightningnetwork/lnd/kvdb"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+var (
+	// announcementNoncesBucket is a top level bucket with the following
+	// structure: channel_id => node_nonce || bitcoin_nonce
+	announcementNoncesBucket = []byte("announcement-nonces")
+)
+
+// AnnouncementNonces holds the nonces used during the creating of a
+// ChannelAnnouncement2.
+type AnnouncementNonces struct {
+	Btc  [musig2.PubNonceSize]byte
+	Node [musig2.PubNonceSize]byte
+}
+
+// SaveAnnouncementNonces persist the given announcement nonces.
+func (c *ChannelStateDB) SaveAnnouncementNonces(chanID lnwire.ChannelID,
+	nonces *AnnouncementNonces) error {
+
+	chanIDCopy := make([]byte, 32)
+	copy(chanIDCopy, chanID[:])
+
+	scratch := make([]byte, musig2.PubNonceSize*2)
+	copy(scratch[:musig2.PubNonceSize], nonces.Node[:])
+	copy(scratch[musig2.PubNonceSize:], nonces.Btc[:])
+
+	return kvdb.Update(c.backend, func(tx kvdb.RwTx) error {
+		bucket, err := tx.CreateTopLevelBucket(
+			announcementNoncesBucket,
+		)
+		if err != nil {
+			return err
+		}
+
+		return bucket.Put(chanIDCopy, scratch)
+	}, func() {})
+}
+
+// GetAllAnnouncementNonces returns all the announcement nonce pairs currently
+// stored in the DB.
+func (c *ChannelStateDB) GetAllAnnouncementNonces() (
```

Comment:
What use is it to be able to fetch all the nonces vs just one for the peer that just reconnected?